hcn/hcnnetwork.go

```diff
@@ -213,6 +221,104 @@ func (w *HcnServiceWatcher) Close() error {
 	return nil
 }
 
+func NewHcnNetworkWatcher() *HcnNetworkWatcher {
+	watcherContext := &notifcationWatcherContext{
+		channel: make(notificationChannel, 1),
+	}
+
+	callbackMapLock.Lock()
+	callbackNumber := nextCallback
+	nextCallback++
+	callbackMap[callbackNumber] = watcherContext
+	callbackMapLock.Unlock()
+
+	return &HcnNetworkWatcher{
+		callbackNumber: callbackNumber,
+		watcherContext: watcherContext,
+	}
+}
+
+func (w *HcnNetworkWatcher) Start(networkId string) error {
+	w.handleLock.RLock()
+	defer w.handleLock.RUnlock()
+
+	if w.started {
+		return nil
+	}
+
+	var networkHandle hcnNetwork
+	var resultBuffer *uint16
+	networkGuid, err := guid.FromString(networkId)
```

Comment:
>err [](http://example.com/codeflow?start=14&length=3)

check if err occurs.

Or consider pass guid type for networkId (if applicable)